[WIP] Lusd bonds

unbanksy opened the pull request.

appleseed-iii committed:
Update views for LUSD Bonds

WIP:
- [ ] SVG for LUSD logo

NEEDS:
- [ ] contract addresses for LUSD bonds on main & rinkeby
- [ ] lusd faucet for testing
- [ ] ohm-lusd stake pair address

appleseed-iii committed:
LUSD BONDS update abi & test deposit function

Deposit function working - OK

Changes:
1. add LUSD Bond abi
2. add LUSD svg logo to bond view
3. update LUSD Rinkeby addresses in AllBonds

appleseed-iii committed:
add ohm-lusd image & prepare for link out to pickle

unbanksy commented on src/constants.js:
```diff
@@ -60,6 +60,7 @@ export const BONDS = {
   dai: "dai",
   frax: "frax",
   eth: "eth",
+  lusd: "eth",
```
Should be `lusd: "lusd"` right?

appleseed-iii replied:
resolved in https://github.com/OlympusDAO/olympus-frontend/pull/453/commits/33f065b7e055949bcf15114632f892c6876e6173

unbanksy commented on src/hooks/web3Context.tsx:
```diff
@@ -106,7 +106,7 @@ export const Web3ContextProvider: React.FC<{ children: ReactElement }> = ({ chil
 
   const [uri, setUri] = useState(getMainnetURI());
-  const [provider, setProvider] = useState<JsonRpcProvider>(new StaticJsonRpcProvider(uri));
+  const [provider, setProvider] = useState<JsonRpcProvider>(new StaticJsonRpcProvider(defaultApiKey));
```
why use this over `uri`?

appleseed-iii replied:
I need to revert that. That was just a temp placeholder during testing.

appleseed-iii replied:
resolved in https://github.com/OlympusDAO/olympus-frontend/pull/453/commits/33f065b7e055949bcf15114632f892c6876e6173

appleseed-iii committed:
Fix lusd-constants reference & fix web3Context permanent api key

appleseed-iii committed:
Add API to Pickle for APY, TVL & get User Balance

WIP: swap out contracts when we have them

Zayen-X commented on src/constants.js:
```diff
@@ -60,6 +64,7 @@ export const BONDS = {
   dai: "dai",
   frax: "frax",
   eth: "eth",
+  lusd: "lusd",
```
Do we still use this object? we _shouldn't_

appleseed-iii replied:
resolved in https://github.com/OlympusDAO/olympus-frontend/pull/453/commits/f226a02fe86094d69c9d25890ab62bc383928afd

Zayen-X commented on src/slices/LusdSlice.js:
```diff
@@ -0,0 +1,67 @@
+import { ethers } from "ethers";
```
Since this is a new file. do you mind using TS for this? :) intentional TypeScipt at the time of writing this is definitely better than retroactively going back and adding objects.

appleseed-iii replied:
Yes, let me work on that.

appleseed-iii replied:
resolved in https://github.com/OlympusDAO/olympus-frontend/pull/453/commits/f226a02fe86094d69c9d25890ab62bc383928afd

Zayen-X commented on src/slices/LusdSlice.js:
```diff
@@ -0,0 +1,67 @@
+import { ethers } from "ethers";
+import { addresses, Actions } from "../constants";
```
What you import `Actions` for?

appleseed-iii replied:
removing this

appleseed-iii replied:
resolved in https://github.com/OlympusDAO/olympus-frontend/pull/453/commits/f226a02fe86094d69c9d25890ab62bc383928afd

Zayen-X commented on src/views/Stake/Stake.jsx:
```diff
@@ -423,31 +426,32 @@ function Stake() {
                         <Box className="ohm-pairs">
-                          <BondLogo bond={ohm_frax}></BondLogo>
-                          <Typography>OHM-FRAX</Typography>
+                          <BondLogo bond={{ bondIconSvg: OhmLusdImg, isLP: true }}></BondLogo>
```
Can you just pass in the bond object?

appleseed-iii replied:
We don't have an Ohm-Lusd bond yet. When we do I'll pass the bond object.

Zayen-X commented on src/views/Stake/Stake.jsx:
```diff
@@ -423,31 +426,32 @@ function Stake() {
+                          }).format(lusdData.tvl)}
                       </TableCell>
-                      <TableCell align="left"> {(fraxData && fraxData.balance) || 0} LP </TableCell>
+                      <TableCell align="left"> {(lusdData && trim(lusdData.balance, 2)) || 0} LP </TableCell>
```
stuff like this makes me realize that we need to work better to generalize this section. kek.

Zayen-X reviewed (commented):
Minor comments, but don't want to block the release.

appleseed-iii committed:
PR Comment fixes & add lusd mainnet contract & abi
1. Add mainnet lusd-bond contract
2. refactor LusdSlice into ts
3. remove stale imports
4. re-tested & verified operation of rinkeby lusd bonds

appleseed-iii committed:
refactor External stake view into a separate component & remove from Ohm Stake View

Zayen-X commented on src/App.jsx:
```diff
@@ -91,7 +91,7 @@ function App() {
   const isAppLoaded = useSelector(state => typeof state.app.marketPrice != "undefined"); // Hacky way of determining if we were able to load app Details.
   const { bonds } = useBonds();
-  async function loadDetails(whichDetails) {
+  async function loadDetails(whichDetails: string) {
```
probs don't need this because its a .jsx

appleseed-iii replied:
My vscode always yelling at me that it needs type declared. That's why I changed it: 
![image](https://user-images.githubusercontent.com/80423742/134185109-db56be9d-ba49-46e1-b3d5-8435a8d78a37.png)

Zayen-X commented on src/views/Stake/ExternalStakePool.jsx:
```diff
@@ -0,0 +1,165 @@
+import { useEffect } from "react";
```
<3

Zayen-X reviewed (approved):
LGTM :D

Zayen-X committed:
remove ts in jsx

Zayen-X committed:
merged develop

Zayen-X commented on src/helpers/AllBonds.ts:
```diff
@@ -1,18 +1,18 @@
+import FraxImg from "src/assets/tokens/FRAX.svg";
+import OhmLusdImg from "src/assets/tokens/OHM-LUSD.svg";
+import wETHImg from "src/assets/tokens/wETH.svg";
+import LusdImg from "src/assets/tokens/LUSD.svg";
```
I think broke something.

appleseed-iii replied:
weird, what happened. I'll fix.

appleseed-iii replied:
fixed svg in https://github.com/OlympusDAO/olympus-frontend/pull/453/commits/8f44220d33b840e46e19d8d238c0eccd2f1e0dee

appleseed-iii committed:
Updates to handle develop branch merge bugs:

1. remove useEffect check for accountBonds since it did nothing
2. update LUSD & OHM-LUSD svgs to work with current develop branch
3. update imports for ALL bond SVGs to work with develop branch (was throwing white screen of death)
